uaclient/actions.py

```diff
@@ -218,3 +219,52 @@ def collect_logs(cfg: config.UAConfig, output_dir: str):
             system.write_file(
                 os.path.join(output_dir, os.path.basename(f)), content
             )
+
+
+def should_disable_auto_attach(cfg: config.UAConfig) -> bool:
+    disable_auto_attach = util.is_config_value_true(
+        config=cfg.cfg, path_to_value="features.disable_auto_attach"
+    )
+    if disable_auto_attach:
+        msg = "Skipping auto-attach. Config disable_auto_attach is set."
+        logging.debug(msg)
+        print(msg)
```

Comment:
This printing shouldn't happen in the api function. This should only happen in the cli module. (sorry I didn't notice this last time I commented on this bit of code :frowning_face:)

Reply:
Yes, that is why I separated it to another function, `shoudl_disable_auto_attach` which is only called by the cli module. We could move this code to the cli module as well, since it is not called by another module.

Reply:
Ah I see - sorry I misunderstood! This looks good, no need to change